platforms/jvm/jacoco/src/main/java/org/gradle/testing/jacoco/plugins/JacocoPlugin.java

```diff
@@ -245,14 +250,15 @@ private void addDefaultReportTask(final JacocoPluginExtension extension, final T
             });
     }
 
-    private void addDefaultCoverageVerificationTask(final TaskProvider<? extends Task> testTaskProvider) {
+    private void addDefaultCoverageVerificationTask(final TaskProvider<? extends Task> testTaskProvider, TaskProvider<? extends JavaCompile> compileJavaTaskProvider) {
         project.getTasks().register(
             "jacoco" + StringUtils.capitalize(testTaskProvider.getName()) + "CoverageVerification",
             JacocoCoverageVerification.class,
             coverageVerificationTask -> {
                 coverageVerificationTask.setGroup(LifecycleBasePlugin.VERIFICATION_GROUP);
                 coverageVerificationTask.setDescription(String.format("Verifies code coverage metrics based on specified rules for the %s task.", testTaskProvider.getName()));
                 coverageVerificationTask.executionData(testTaskProvider.get());
+                coverageVerificationTask.getSourceEncoding().set(compileJavaTaskProvider.flatMap(compileTask -> project.getProviders().provider(() -> compileTask.getOptions().getEncoding())));
```

Comment:
Why `flatMap` into `provider(() -> )` instead of `map` without the provider?

Reply:
Because `map` would preserve the task-dependency and I do not see a reason to have the coverage verification task depend on the compile java task just because I want to use the same encoding that was configured there.